src/js/components/Topic/Index.jsx

```diff
@@ -58,21 +82,95 @@ export default class Index extends React.Component {
-                                    </label>
-                                </div>
-                            </div>
-                        </form>
-                        {topics}
-                    </div>
+                    <div className="small-3 columns">
+                    <h4>Filter</h4>
+                    <hr/>
+                    <label>
+                        <input
+                            type="text"
+                            defaultValue={filterCriteria.search}
+                            onChange={(e) => { this.search(e); }}
+                            placeholder="Search"/>
+                    </label>
+                </div>
+                <div className="small-9 columns">
+                <h4>{data.length} topic(s)</h4>
+                <hr/>
+
+                <TopicListSortOrder
+                    filter={(key, value) => this.setFilter(key, value)}
+                    activeSortOrder={filterCriteria.order} />
+
+                {topics}
                 </div>
             </div>
+        </div>
         );
     }
-}
\ No newline at end of file
+
+    filterTopics(topics, criteria) {
+        return topics.filter((topic) => {
+            var search = criteria.search;
+
+            if (search != undefined && search.length > 0 && ! topic.title.toLowerCase().includes(search.toLowerCase())) {
+                return false;
+            }
+
+            return true;
+        });
+    }
+
+    sortTopics(topics, sortOrder) {
+        let column = 'title';
+        let direction = 'asc';
+        let length = false;
+
+        switch (sortOrder) {
+            case 'title_asc':
+                column = 'title';
+                break;
+            case 'title_desc':
+                column = 'title';
+                direction = 'desc';
+                break;
+            case 'votes':
+                column = 'votes';
+                direction = 'desc';
+                length = true;
+                break;
+            case 'newest':
+                column = 'createdAt';
+                direction = 'desc';
+                break;
+            case 'oldest':
+                column = 'createdAt';
+                direction = 'asc';
+                break;
+        }
+
+        console.log('topic', topics[0]);
+        return topics.sort((topicA, topicB) => {
+            let valueA = topicA[column];
+            let valueB = topicB[column];
+
+            if (length) {
+                valueA = valueA.length;
+                valueB = valueB.length;
+            }
+
+            if (valueA > valueB) {
+                return direction == 'asc'
+                    ? 1
+                    : -1;
+            }
+
+            if (valueA < valueB) {
+                return direction == 'asc'
+                    ? -1
+                    : 1;
+            }
+
+            return 0;
```

Comment:
Wouldn't `return valueA > valueB``return boolean true|false whereas sort functions need 0|1|-1? Not sure how JS handles this exactly ...